hw/mcu/stm/stm32f3xx/src/hal_gpio.c

```diff
@@ -187,69 +187,69 @@ hal_gpio_clk_enable(uint32_t port_idx)
                 __HAL_RCC_GPIOB_CLK_ENABLE();
             }
             break;
+#endif
 #if HAL_GPIO_PORT_COUNT > 2
         case 2:
             if (!__HAL_RCC_GPIOC_IS_CLK_ENABLED()) {
                 __HAL_RCC_GPIOC_CLK_ENABLE();
             }
             break;
+#endif
 #if HAL_GPIO_PORT_COUNT > 3
         case 3:
             if (!__HAL_RCC_GPIOD_IS_CLK_ENABLED()) {
                 __HAL_RCC_GPIOD_CLK_ENABLE();
             }
             break;
+#endif
 #if HAL_GPIO_PORT_COUNT > 4 && defined GPIOE_BASE
```

Comment:
Cool catching this! I would just have changed this line to:

```
#if HAL_GPIO_PORT_COUNT > 4
  #if defined(GPIOE_BASE)
  ...
  #endif
#if HAL_GPIO_PORT_COUNT > 5
```
I believe it would fix the same issue without moving all those `#endif` lines around.